### PR TITLE
Operative completion

### PR DESCRIPTION
**Pull request by josh-dadswell-hackney**
Allow operatives to complete work orders they are assigned to
https://www.pivotaltracker.com/story/show/179282621

### CODE REVIEW

File: RepairsApi.Tests/E2ETests/Repairs/RepairApiTests.cs [hunk 1]
```diff
@@ -239,6 +239,24 @@ public async Task CompleteWorkOrder()
...
+        [Test]
+        public async Task OperativeCompleteWorkOrder()
+        {
+            // Arrange
+            var eventTime = DateTime.Now.AddDays(-2);
+            var result = await CreateWorkOrder();
+            await AssignOperative(result.Id, TestDataSeeder.OperativeId);
+            SetUserRole(UserGroups.Operative);
+
+            var code = await CompleteWorkOrder(result.Id, true, woc => woc.ClosedTime = eventTime);
```
Comment (Asura5) [inline RepairsApi.Tests/E2ETests/Repairs/RepairApiTests.cs:251]: Add the Act comment if you're going to have the comments

Comment (josh-dadswell-hackney) [reply RepairsApi.Tests/E2ETests/Repairs/RepairApiTests.cs:247]: Tidied the whole file

File: RepairsApi/V2/Infrastructure/Migrations/20210901115433_OperativeEmailUnique.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace RepairsApi.V2.Infrastructure.Migrations
+{
+    public partial class OperativeEmailUnique : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateIndex(
```
Comment (Asura5) [inline RepairsApi/V2/Infrastructure/Migrations/20210901115433_OperativeEmailUnique.cs:9]: Given that operatives is a smallish table, have we considered the performance overhead?

File: RepairsApi/V2/UseCase/CompleteWorkOrderUseCase.cs [hunk 3]
```diff
@@ -117,21 +114,38 @@ private async Task HandleCustomType(WorkOrder workOrder, JobStatusUpdate update)
...
+                !_currentUserService.HasGroup(UserGroups.ContractManager) &&
+                !IsAssignedOperative(workOrder))
                 throw new UnauthorizedAccessException("Not Authorised to close jobs");
         }
 
+        private bool IsAssignedOperative(WorkOrder workOrder)
+        {
+            var user = _currentUserService.GetUser();
+            return _currentUserService.HasGroup(UserGroups.Operative) &&
+                   workOrder.AssignedOperatives.Exists(ao => string.Equals(ao.PayrollNumber, user.PayrollNumber(), StringComparison.InvariantCultureIgnoreCase));
```
Comment (Asura5) [inline RepairsApi/V2/UseCase/CompleteWorkOrderUseCase.cs:145]: DRY - This is also in RepairsGateway.cs - consider refactor out to a helper method

**Review by Asura5 — COMMENTED** [review]
A few individual items - one overall comment is that in future please try to keep the PR size reasonable - splitting the auth and completion logic out for example

File: RepairsApi/V2/Infrastructure/Migrations/20210901115433_OperativeEmailUnique.cs (hunk 2 above)
Comment (josh-dadswell-hackney) [reply RepairsApi/V2/Infrastructure/Migrations/20210901115433_OperativeEmailUnique.cs:9]: I dint understand your concern, if its small and should only impact on insert I dont see the problem. please elaborate.

File: RepairsApi/V2/UseCase/CompleteWorkOrderUseCase.cs (hunk 3 above)
Comment (josh-dadswell-hackney) [reply RepairsApi/V2/UseCase/CompleteWorkOrderUseCase.cs]: Good shout

**Review by Asura5 — APPROVED** [review]

**Pull request merged**